platform/posix/transport/src/openssl_posix.c

```diff
@@ -607,12 +606,26 @@ int32_t Openssl_Recv( NetworkContext_t * pNetworkContext,
     int32_t bytesReceived = 0;
     int sslError = 0;
 
-    /* SSL read of data. */
-    bytesReceived = ( int32_t ) SSL_read( pNetworkContext->pSsl,
-                                          pBuffer,
-                                          bytesToRecv );
+    if( pNetworkContext == NULL )
+    {
+        LogError( ( "Parameter check failed: pNetworkContext is NULL." ) );
+    }
+    else if( pNetworkContext->pSsl != NULL )
+    {
+        /* SSL read of data. */
+        bytesReceived = ( int32_t ) SSL_read( pNetworkContext->pSsl,
+                                              pBuffer,
+                                              bytesToRecv );
+    }
```

Comment:
It looks like this code can be further simplified – move lines 628-641 here (between lines 618/619) and delete the first two tests at line 627, e.g.:

```c
if( bytesReceived <= 0 )
{
    sslError = SSL_get_error( pNetworkContext->pSsl, bytesReceived );
    if( sslError == SSL_ERROR_WANT_READ )
    {
        /* There is no data to receive at this time. */
        bytesReceived = 0;
    }
    else
    {
        LogError( ( "Failed to receive data over network: SSL_read failed: "
                        "ErrorStatus=%s.", ERR_reason_error_string( sslError ) ) );
     }
}

Reply:
Moving these lines within the if block of `if( pNetworkContext->pSsl != NULL )` (line  613) will cause 3 levels of nesting, and therefore, I suspect that the complexity score would cross our threshold of 8...

Reply:
Interestingly, the complexity tool didn't complain about the 3 levels of nesting. So have moved the error handling logic within the `if( pNetwork->pSsl != NULL` block!

Reply:
The suggested change reduces the McCabe score of `Openssl_Recv()` from 7 to 5, see results below.

__Before Suggested Change__

```sh
$ pmccabe -v platform/posix/transport/src/openssl_posix.c 
Modified McCabe Cyclomatic Complexity
|   Traditional McCabe Cyclomatic Complexity
|       |    # Statements in function
|       |        |   First line of function
|       |        |       |   # lines in function
|       |        |       |       |  filename(definition line number):function
|       |        |       |       |           |
4	4	9	131	30	platform/posix/transport/src/openssl_posix.c(131): logPath
10	10	27	164	67	platform/posix/transport/src/openssl_posix.c(164): setRootCa
3	3	10	233	29	platform/posix/transport/src/openssl_posix.c(233): setClientCertificate
3	3	10	264	30	platform/posix/transport/src/openssl_posix.c(264): setPrivateKey
6	6	10	296	30	platform/posix/transport/src/openssl_posix.c(296): setCredentials
8	8	19	328	65	platform/posix/transport/src/openssl_posix.c(328): setOptionalConfigurations
23	23	63	395	167	platform/posix/transport/src/openssl_posix.c(395): Openssl_Connect
5	5	13	564	36	platform/posix/transport/src/openssl_posix.c(564): Openssl_Disconnect
7	7	13	602	44	platform/posix/transport/src/openssl_posix.c(602): Openssl_Recv
4	4	11	648	34	platform/posix/transport/src/openssl_posix.c(648): Openssl_Send
```

__After Suggested Change__

```sh
Modified McCabe Cyclomatic Complexity
|   Traditional McCabe Cyclomatic Complexity
|       |    # Statements in function
|       |        |   First line of function
|       |        |       |   # lines in function
|       |        |       |       |  filename(definition line number):function
|       |        |       |       |           |
4	4	9	131	30	platform/posix/transport/src/openssl_posix.c(131): logPath
10	10	27	164	67	platform/posix/transport/src/openssl_posix.c(164): setRootCa
3	3	10	233	29	platform/posix/transport/src/openssl_posix.c(233): setClientCertificate
3	3	10	264	30	platform/posix/transport/src/openssl_posix.c(264): setPrivateKey
6	6	10	296	30	platform/posix/transport/src/openssl_posix.c(296): setCredentials
8	8	19	328	65	platform/posix/transport/src/openssl_posix.c(328): setOptionalConfigurations
23	23	63	395	167	platform/posix/transport/src/openssl_posix.c(395): Openssl_Connect
5	5	13	564	36	platform/posix/transport/src/openssl_posix.c(564): Openssl_Disconnect
5	5	13	602	41	platform/posix/transport/src/openssl_posix.c(602): Openssl_Recv
4	4	11	645	34	platform/posix/transport/src/openssl_posix.c(645): Openssl_Send
```